### PR TITLE
Add/stm32mp157a dk1

### PR DESCRIPTION
**Pull request by jorisoffouga**

### CODE REVIEW

**Comment by jorisoffouga** [comment]
See raw log : 
root@stm32mp157a-dk1:~# swupdate -v -i update-image-stm32mp157a-dk1.swu --select=stable,rootfsB -H stm32mp157a-dk1:rev0.1 -f /etc/swupdate.cfg
Swupdate v2019.04.0

Licensed under GPLv2. See source distribution for detailed copyright notices.

Running on stm32mp157a-dk1 Revision rev0.1
Registered handlers:
	dummy
	archive
	tar
	uboot
	bootloader
	lua
	raw
	rawfile
	remote
	shellscript
	preinstall
	postinstall
	swuforward
software set: stable mode: rootfsB
[TRACE] : SWUPDATE running :  [network_initializer] : Main loop Daemon
[TRACE] : SWUPDATE running :  [listener_create] : got no socket at /tmp/swupdateprog from systemd
[TRACE] : SWUPDATE running :  [listener_create] : creating socket at /tmp/swupdateprog
[TRACE] : SWUPDATE running :  [listener_create] : got no socket at /tmp/sockinstctrl from systemd
[TRACE] : SWUPDATE running :  [listener_create] : creating socket at /tmp/sockinstctrl
[TRACE] : SWUPDATE running :  [extract_sw_description] : Found file:
	filename sw-description
	size 1160
	checksum 0x112a1 VERIFIED
[TRACE] : SWUPDATE running :  [get_common_fields] : Version 1.0.0
[TRACE] : SWUPDATE running :  [parse_hw_compatibility] : Accepted Hw Revision : rev0.1
[TRACE] : SWUPDATE running :  [parse_images] : Found compressed Image: base-image-swupdate-stm32mp157a-dk1.ext4.gz in device : /dev/mmcblk0p6 for handler raw (installed from stream)
[TRACE] : SWUPDATE running :  [parse_bootloader] : Bootloader var: rootpart = 6
[TRACE] : SWUPDATE running :  [parse_bootloader] : Bootloader var: upgrade_available = 1
[TRACE] : SWUPDATE running :  [parse_bootloader] : Bootloader var: bootcount = 0
[TRACE] : SWUPDATE running :  [check_hw_compatibility] : Hardware stm32mp157a-dk1 Revision: rev0.1
[TRACE] : SWUPDATE running :  [check_hw_compatibility] : Hardware compatibility verified
[TRACE] : SWUPDATE running :  [cpio_scan] : Found file:
	filename base-image-swupdate-stm32mp157a-dk1.ext4.gz
	size 37003925
	REQUIRED
[TRACE] : SWUPDATE running :  [install_single_image] : Found installer for stream base-image-swupdate-stm32mp157a-dk1.ext4.gz raw
Software updated successfully
Please reboot the device to start the new software
[INFO ] : SWUPDATE successful ! 
[DEBUG] : SWUPDATE running :  [postupdate] : Executing post-update command 'reboot'
[DEBUG] : SWUPDATE running :  [postupdate] : Post-update command returned 0

File: wic/swu-stm32mp1.wks [hunk 1]
```diff
@@ -0,0 +1,9 @@
+part fsbl1 --source rawcopy --sourceparams="file=u-boot-spl.stm32" --ondisk mmcblk --align 1 --size 256k --part-name fsbl1
+part fsbl2 --source rawcopy --sourceparams="file=u-boot-spl.stm32" --ondisk mmcblk --align 1 --size 256k --part-name fsbl2
+part u-boot --source rawcopy --sourceparams="file=u-boot.img" --ondisk mmcblk --align 1 --size 2M --part-name uboot
+part /boot --source bootimg-partition --ondisk mmcblk --fstype=vfat --label boot --active --align 1 --size 20
+part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_A  --align 1 --extra-space 200
+part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_B  --align 1 --extra-space 200
+part /data --ondisk mmcblk --fstype=ext4 --label data --size 128M --align 1 --extra-space 0
+
+bootloader --ptable gpt
```
Comment (texierp) [inline wic/swu-stm32mp1.wks]: Please add a new line here

File: recipes-bsp/libubootenv/libubootenv/stm32mp157a-dk1/fw_env.config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/dev/mmcblk0 0x280000 0x2000
+/dev/mmcblk0 0x300000 0x2000
```
Comment (texierp) [inline recipes-bsp/libubootenv/libubootenv/stm32mp157a-dk1/fw_env.config]: Please add a new line here

File: recipes-support/swupdate/swupdate/stm32mp157a-dk1/swupdate.cfg [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+download :
+{
+	retries = 3;
+	timeout = 1800;
+};
+
+identify : (
+	{ name = "stm32mp157a-dk1"; value = "rev0.1"; }
```
Comment (texierp) [inline recipes-support/swupdate/swupdate/stm32mp157a-dk1/swupdate.cfg]: U-Boot gives me Rev.C-01

```
STM32MP> stboard 
Board: MB1272 Var1 Rev.C-01
```

File: recipes-bsp/u-boot-scr/files/boot.cmd.in [hunk 4]
```diff
@@ -1,4 +1,4 @@
-setenv bootargs 'console=${console},${baudrate} root=@@BLK_DEVICE@@p'${rootpart}' rootwait rw'
+setenv bootargs 'console=${console},${baudrate} root=@@BLK_DEVICE@@p'${rootpart}' oops=panic panic=5 rootwait rw'
```
Comment (texierp) [inline recipes-bsp/u-boot-scr/files/boot.cmd.in]: `console` is not define in U-Boot env for the stm32.

I would recommend:

```
if env exists console
then 
	setenv bootargs 'console=${console},${baudrate} root=@@BLK_DEVICE@@p'${rootpart}' oops=panic panic=5 rootwait rw'
else 
	setenv bootargs 'root=@@BLK_DEVICE@@p'${rootpart}' oops=panic panic=5 rootwait rw'
fi
```

File: conf/site.conf [hunk 5]
```diff
@@ -47,7 +56,13 @@ DISTRO_FEATURES_remove = " \
 "
 
 # Keep libubootenv and U-Boot
-RM_WORK_EXCLUDE = "libubootenv u-boot-fslc swupdate"
+RM_WORK_EXCLUDE = "libubootenv virtual/bootloader swupdate"
 
+# BLK for U-Boot script
```
Comment (texierp) [inline conf/site.conf]: 'defines block devices for U-Boot boot script'

File: kas.yml [hunk 6]
```diff
@@ -37,6 +37,10 @@ repos:
     url: https://github.com/sbabic/meta-swupdate
     refspec: d345ba975dc72e1674b2356ce01c50ac8f320f37
 
+  meta-stm32mp1:
+    url: https://github.com/bdx-iot/meta-stm32mp1
+    refspec: 4629412d0380222353f4410b99cff2a80a196f44
```
Comment (texierp) [inline kas.yml]: Update to 907cef70f85cfb8036ec93f61213d15c40388b38

File: recipes-bsp/u-boot/u-boot-mainline/stm32mp1/0001-stm32mp1-add-SWUpdate-Requirements.patch [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+--- a/configs/stm32mp15_basic_defconfig
++++ b/configs/stm32mp15_basic_defconfig
+@@ -9,6 +9,10 @@ CONFIG_SPL_SPI_SUPPORT=y
+ CONFIG_DISTRO_DEFAULTS=y
+ CONFIG_FIT=y
+ CONFIG_BOOTCOMMAND="run bootcmd_stm32mp"
++CONFIG_BOOTCOUNT_LIMIT=y
++CONFIG_BOOTCOUNT_ENV=y
++CONFIG_USE_PREBOOT=y
++CONFIG_PREBOOT="if env exists swu_setup; then echo 'Use default'; else env save; env save;fi"
```
Comment (texierp) [inline recipes-bsp/u-boot/u-boot-mainline/stm32mp1/0001-stm32mp1-add-SWUpdate-Requirements.patch:20]: Have you sync the defconfig with `savedefconfig` ?

**Comment by texierp** [comment]
Reviewed-by: Pierre-Jean Texier <pjtexier@koncepto.io>

**Review by texierp — APPROVED** [review]

**Pull request merged**